service/src/main/java/tds/config/services/ConfigService.java

```diff
@@ -16,4 +16,17 @@
      * @return The {@link Optional<ClientSystemFlag>} that matches the client and audit name.
      */
     Optional<ClientSystemFlag> findClientSystemFlag(String clientName, String type);
+
+    /**
+     * Get the message string for this client that is translated for the language specified.
+     *
+     * @param clientName The client name
+     * @param messageKey The message key found in the {@code configs.tds_coremessageobject} table in the {@code appkey} column
+     * @param language   The language code to translate the message into
+     * @param context    The context for the message found in the {@code configs.tds_coremessageobject} table in the {@code context} column
+     * @param subject    A subject code used to find a more specific message.  NULL will match on all
+     * @param grade      A grade level used to find a more specific message.  NULL will match on all
+     * @return The message string with placeholders included
+     */
+    String getSystemMessage(String clientName, String messageKey, String language, String context, String subject, String grade);
```

Comment:
Any thoughts of breaking the message stuff into its own service rather than combining with this service?   Same for the repository.  Keeps the services and repositories simple and focused on a single responsibility.